src/lib/stitching/convection/v2/stitching.ts

```diff
@@ -90,5 +100,45 @@ export const consignmentStitchingEnvironment = (
           ),
       },
     },
+    Mutation: {
+      createConsignmentSubmission: {
+        resolve: async (_source, args, context, info) => {
+          const myCollectionArtworkID = args.input?.myCollectionArtworkID
+          let createSubmissionArgs = args
+
+          // when myCollectionArtworkID is specified, use artwork data to fill in submission
+          if (myCollectionArtworkID) {
+            if (context.artworkLoader) {
+              const artwork = await context.artworkLoader(myCollectionArtworkID)
+
+              if (!artwork) {
+                throw new Error("Artwork not found")
+              }
+
+              const artworkSubmissionData = artworkToSubmissionMapping(artwork)
```

Comment:
We need to also specify that the source  is `MY_COLLLECTION` when the artwork comes from my collection.

Reply:
Good idea! I wonder though whether we should set it in the resolver if myCollectionArtworkId is present or if the client should pass it explicitly as an argument 🤔

Reply:
I feel like if we are creating an artwork using a my collection artwork id, we can confidently tell this artwork source is my collection.